src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -824,22 +818,17 @@ public IEnumerable<Document> GetDocuments(Solution solution)
                                 var project = solution.GetProject(projectId);
                                 if (project != null)
                                 {
-                                    foreach (var document in project.Documents)
-                                    {
-                                        yield return document;
-                                    }
+                                    foreach (var documentId in project.DocumentIds)
+                                        yield return (project, documentId);
```

Comment:
Note this is code _may_ get updated for source generator stuff; going to be looking at this later today but not sure if this bit will stick.

Reply:
k.